src/Surfnet/StepupMiddleware/ApiBundle/Controller/CommandController.php

```diff
@@ -19,22 +19,36 @@
 namespace Surfnet\StepupMiddleware\ApiBundle\Controller;
 
 use Surfnet\StepupMiddleware\CommandHandlingBundle\Command\Command;
+use Surfnet\StepupMiddleware\CommandHandlingBundle\Exception\ForbiddenException;
 use Surfnet\StepupMiddleware\CommandHandlingBundle\Pipeline\Pipeline;
 use Symfony\Bundle\FrameworkBundle\Controller\Controller;
 use Symfony\Component\HttpFoundation\JsonResponse;
 use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\HttpException;
 
 class CommandController extends Controller
 {
     public function handleAction(Command $command, Request $request)
     {
+        /** @var \Monolog\Logger $logger */
+        $logger = $this->get('logger');
+
+        $logger->notice(sprintf('Received request to process Command "%s"', $command));
+
         /** @var Pipeline $pipeline */
         $pipeline = $this->get('pipeline');
-        $command = $pipeline->process($command);
+
+        try {
+            $command = $pipeline->process($command);
+        } catch (ForbiddenException $e) {
+            throw new HttpException(403, sprintf('Processing of command "%s" is forbidden for this client', $command));
```

Comment:
Effectively the same, however it is not that access is denied (you are authenticated and are accessing the resource).... that said, going to change it anyway as frankly it doesn't really matter much :wink: